src/Button/Button.tsx

```diff
@@ -2,22 +2,26 @@ import React, { FC, ReactNode } from 'react'
 import styled, { css } from 'styled-components'
 
 import { theme } from '../theme'
+import { LegacyButton } from './LegacyButton'
+import { Loader } from '../Loader'
+import { Icon as IconComponent } from '../Icon'
+import { FontStyle } from '../fontStyle'
 
 interface IButton {
-  /** button color  */
-  color: string
-  /** unique id */
-  id: string
-  /** take full 100% width  */
-  block: boolean
-  /** invert bg and text colors */
-  inverted: boolean
   /** disabled state */
   disabled: boolean
-  /** outline styling */
-  outlined: boolean
-  /** onClick event handler */
-  onClick: (e: React.FormEvent<HTMLButtonElement>) => void
+  /** loading state */
+  loading: boolean
+  /** primary button styling */
+  primary: boolean
+  /** secondary button styling */
+  secondary: boolean
+  /** tertiary button styling */
+  tertiary: boolean
+  /** icon state */
+  icon: string
+  /** forcedWidth styling */
+  forcedWidth: string
 }
 
 type Props = {
```

Comment:
Do you think it's worth being more explicit with these names? ie. `ButtonProps` rather than `Props` or is it overkill. We have different patterns in the rest of the codebase and it would be nice to agree an approach.

Reply:
No I agree, have changed to ButtonProps and we should stick to this